hashicorp/boundary-ui
# refactor: 💡 Migrating to `Hds::Table` in OIDC auth-method form

✅ Closes: https://hashicorp.atlassian.net/browse/ICU-11719

:tickets: [Jira ticket](https://hashicorp.atlassian.net/browse/ICU-11719)

## Description
- Migrating field tables to `Hds::Table` from `Rose::Table` in OIDC auth method form.
- The `Form::Field::ListWrapper` doesn't support a `select` yield, so we are creating the table based on some patterns that are in used in the `form/auth-method/ldap` [template](https://github.com/hashicorp/boundary-ui/blob/8ad4e93ff7ccba6c25423089a7648e1f659aa6d6/ui/admin/app/components/form/auth-method/ldap/index.hbs#L385)
- Migrated `Rose::Buttons` to `Hds::Buttons`



## Screenshots (if appropriate):
### Before - OIDC new form:
<img width="1684" alt="image" src="https://github.com/hashicorp/boundary-ui/assets/24277002/c4712f9b-eb7f-4de6-be48-bdeb9599066c">

### After - OIDC new form:
<img width="1684" alt="image" src="https://github.com/hashicorp/boundary-ui/assets/24277002/04e4a508-078d-499f-a2a2-dae0110b2800">

### Before - OIDC detail view:
<img width="1684" alt="image" src="https://github.com/hashicorp/boundary-ui/assets/24277002/614ab4a5-1ce4-45d5-ab82-e91a00aaccd7">

### After - OIDC detail view:
<img width="1684" alt="Screenshot 2024-01-12 at 10 16 08" src="https://github.com/hashicorp/boundary-ui/assets/24277002/041abc9f-0100-4c9e-adf1-bb28cb5ad799">

## How to Test
- Click on Vercel link for Admin UI or pull down PR and run it locally using `yarn start`
- Click on "Auth Method" in the sidebar navigation.
- Click on the "New" button in the top right hand corner of page and select OIDC.
- Ensure in the "Signing Algorithms" section you are able to select the different algorithms and add them to the table.
- Ensure in the "Account Claim Maps" section you are able to input and select the key-value pairs and add them to the table.
- After saving the form ensure that you are able to go back to the created OIDC and see that data in a disabled state.





:technologist: [Admin preview](PATH_TO_FEATURE)

## Checklist:


~~- [ ] I have added JSON response output for API changes~~
~~- [ ] I have added steps to reproduce and test for bug fixes in the description~~
~~- [ ] I have commented on my code, particularly in hard-to-understand areas~~
~~- [ ] I have added tests that prove my fix is effective or that my feature works~~
